Author: Daniel Nicoletti <dantti12@gmail.com>
Date:   Mon Apr 1 11:37:18 2013 -0300

    Remove Q_DECLARES that are now on the new pk-qt

diff --git a/ApperKCM/CategoryModel.h b/ApperKCM/CategoryModel.h
index 435145a..4bf85f6 100644
--- a/ApperKCM/CategoryModel.h
+++ b/ApperKCM/CategoryModel.h
@@ -70,6 +70,4 @@ private:
     QModelIndex  m_rootIndex;
 };

-Q_DECLARE_METATYPE(PackageKit::Transaction::Group)
-
 #endif
diff --git a/ApperKCM/Updater/UpdateDetails.cpp b/ApperKCM/Updater/UpdateDetails.cpp
index 7094ec5..53cf7b2 100644
--- a/ApperKCM/Updater/UpdateDetails.cpp
+++ b/ApperKCM/Updater/UpdateDetails.cpp
@@ -146,6 +146,8 @@ void UpdateDetails::hide()

 void UpdateDetails::display()
 {
+    kDebug() << sender();
+
     // set transaction to 0 as if PK crashes
     // UpdateDetails won't be emmited
     m_transaction = 0;
@@ -162,6 +164,8 @@ void UpdateDetails::display()

         m_fadeDetails->setDirection(QAbstractAnimation::Forward);
         m_fadeDetails->start();
+    } else if (m_currentDescription.isEmpty()) {
+        updateDetailFinished();
     }
 }

diff --git a/apperd/TransactionJob.cpp b/apperd/TransactionJob.cpp
index b9de911..4249780 100644
--- a/apperd/TransactionJob.cpp
+++ b/apperd/TransactionJob.cpp
@@ -29,8 +29,6 @@

 #include <KDebug>

-Q_DECLARE_METATYPE(PackageKit::Transaction::Error)
-
 TransactionJob::TransactionJob(Transaction *transaction, QObject *parent) :
     KJob(parent),
     m_transaction(transaction),
diff --git a/apperd/TransactionWatcher.cpp b/apperd/TransactionWatcher.cpp
index b6e93da..31f11d6 100644
--- a/apperd/TransactionWatcher.cpp
+++ b/apperd/TransactionWatcher.cpp
@@ -40,8 +40,6 @@

 #include <Daemon>

-Q_DECLARE_METATYPE(Transaction::Error)
-
 TransactionWatcher::TransactionWatcher(bool packagekitIsRunning, QObject *parent) :
     QObject(parent),
     m_inhibitCookie(-1)
diff --git a/apperd/TransactionWatcher.h b/apperd/TransactionWatcher.h
index 3857fba..81a4307 100644
--- a/apperd/TransactionWatcher.h
+++ b/apperd/TransactionWatcher.h
@@ -67,6 +67,4 @@ private:
     KUiServerJobTracker *m_tracker;
 };

-Q_DECLARE_METATYPE(PackageKit::Transaction::Restart)
-
 #endif
diff --git a/plasmoid/updaterapplet.cpp b/plasmoid/updaterapplet.cpp
index a9548e6..485df95 100644
--- a/plasmoid/updaterapplet.cpp
+++ b/plasmoid/updaterapplet.cpp
@@ -104,8 +104,6 @@ UpdaterApplet::~UpdaterApplet()
 {
 }

-Q_DECLARE_METATYPE(PackageKit::Transaction::Status)
-
 QGraphicsWidget *UpdaterApplet::graphicsWidget()
 {
     if (!m_declarativeWidget) {
